src/main/scala/org/broadinstitute/dsde/firecloud/EntityClient.scala

```diff
@@ -366,6 +370,22 @@ class EntityClient (requestContext: RequestContext, modelSchema: ModelSchema, go
     }
   }
 
+  // ask Rawls if we have permission on this workspace. For writes, Rawls will also check if the workspace is locked.
+  private def validateUpsertPermissions(userInfo: UserInfo, workspaceNamespace: String, workspaceName: String, action: String = "read")(op: Boolean => Future[PerRequestMessage]) = {
```

Comment:
do we have a good reason for this function being a loan pattern instead of just returning a future?